src/main/java/nextstep/subway/applicaion/LineService.java

```diff
@@ -3,78 +3,88 @@
 import nextstep.subway.applicaion.dto.LineRequest;
 import nextstep.subway.applicaion.dto.LineResponse;
 import nextstep.subway.applicaion.dto.SectionRequest;
+import nextstep.subway.applicaion.dto.SectionResponse;
 import nextstep.subway.domain.Line;
 import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Section;
 import nextstep.subway.domain.Station;
+import nextstep.subway.exception.DuplicateException;
+import nextstep.subway.exception.NotFoundLineException;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 import java.util.stream.Collectors;
 
-@Service
 @Transactional
+@Service
 public class LineService {
-    private LineRepository lineRepository;
-    private StationService stationService;
 
-    public LineService(LineRepository lineRepository, StationService stationService) {
+    private final LineRepository lineRepository;
+    private final StationService stationService;
+
+    public LineService(final LineRepository lineRepository, final StationService stationService) {
         this.lineRepository = lineRepository;
         this.stationService = stationService;
     }
 
-    public LineResponse saveLine(LineRequest request) {
-        Line line = lineRepository.save(new Line(request.getName(), request.getColor()));
-        if (request.getUpStationId() != null && request.getDownStationId() != null && request.getDistance() != 0) {
-            Station upStation = stationService.findById(request.getUpStationId());
-            Station downStation = stationService.findById(request.getDownStationId());
-            line.addSection(upStation, downStation, request.getDistance());
+    public LineResponse saveLine(final LineRequest request) {
+        if (isDuplicatedLineName(request.getName())) {
+            throw new DuplicateException();
         }
-        return LineResponse.of(line);
-    }
 
-    @Transactional(readOnly = true)
-    public List<Line> findLines() {
-        return lineRepository.findAll();
-    }
+        Station upStation = stationService.findStationById(request.getUpStationId());
+        Station downStation = stationService.findStationById(request.getDownStationId());
 
-    public List<LineResponse> findLineResponses() {
-        return findLines().stream()
-                .map(LineResponse::of)
-                .collect(Collectors.toList());
+        Line line = Line.of(request.toEntity(), upStation, downStation, request.getDistance());
```

Comment:
라인 정적 팩터리 메서드한테 다시 같은 타입의 객체인 request.toEntity()를 전달하는 것보다 객체 생성에 필요한 값만 전달하면 되지 않을까요?

```suggestion
        Line line = request.toEntity(upStation, downStation, request.getDistance());
```
이렇게 바꿔보면 어떨까요?

Reply:
음 Line.of 내부에 첫 번째 구간을 등록하는 `newLine.sections.addFirst(Section.of(newLine, upStation, downStation, distance));`  코드가 존재합니다.

만약에,  `request.toEntity(upStation, downStation, request.getDistance());` 이렇게 리팩토링을 하게되면, Line 엔티티 생성 규칙(노선과 첫 구간 등록)을 DTO 에서 담당하기 때문에

__엔티티 생성 규칙이 DTO에 의존__ 한다고 생각이 드는데요,,

현구님 리뷰를 참고해서, Line.of 에 같은 타입의 객체를 넘기는 것 말고, 아래와 같이 리팩토링 하는 것에 대해서는 어떻게 생각하시나요?

```java
Line.of(request.getName(), request.getColor(), upStation, downStation, request.getDistance());
```

Reply:
네 말씀하신 방법이 더 좋아보이네요